sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -464,6 +465,9 @@ public BundleFinalizer getBundleFinalizer() {
                 request.getInstructionId(),
                 bundleProcessor.getInboundEndpointApiServiceDescriptors(),
                 observer);
+            if (request.getProcessBundle().hasElements()) {
+              observer.accept(request.getProcessBundle().getElements());
```

Comment:
Wasn't the contract that all the data had to come from the ProcessBundleRequest and if not that would be an error or did we swap back to treating it like a prefix (you have implemented the latter)?